src/nnet3/nnet-simple-component.cc

```diff
@@ -143,7 +143,9 @@ void DropoutComponent::Backprop(const std::string &debug_info,
 
   KALDI_ASSERT(in_value.NumRows() == out_deriv.NumRows() &&
                in_value.NumCols() == out_deriv.NumCols());
-  in_deriv->AddMatMatDivMat(out_deriv, out_value, in_value);
+  CuMatrix<BaseFloat> in_deriv_inplace(in_deriv->NumRows(), in_deriv->NumCols());
```

Comment:
I added kBackpropAdds and kLinearInInput to Properties(). 
It seems AddMatMatDivMat method doesn't add this to its output, which is against what we expect in most of AddMat functions.
I didn't modify this function and I supported kBackpropAdds in backprop by having temporary matrix and adding it to in_deriv!

we may need to modify AddMatMatDivMat method to do *this += a * b / c !!